(startLine=1383 endLine=1397 srcPath=/home/sonia/NewExperiment/guavaFilter/00697/guava-tests/test/com/google/common/util/concurrent/FuturesTest.java)
  public void testCatching_getThrowsError() throws Exception {
    // This is an evil way to accomplish this, but it works currently.
    AbstractFuture<Object> input =
        new AbstractFuture<Object>() {
          @Override
          public Object get() {
            throw new MyError();
          }
        };

    // We'd catch only MyError.class here, but then the test won't compile under GWT.
    ListenableFuture<Object> output = Futures.catching(input, Throwable.class, identity());
    input.setException(new MyException());
    assertThat(output.get()).isInstanceOf(MyError.class);
  }

